File: java-backend/src/main/java/com/ikunmanager/model/Employee.java
package com.ikunmanager.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;
import javax.persistence.*;
import java.util.Date;

@Data
@Entity
@Table(name = "employee")
public class Employee {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @JsonProperty("emp_id")
    @Column(name = "emp_id", unique = true, nullable = false)
    private String empId;

    @Column(nullable = false)
    private String name;

    private String gender;

    private Integer age;

    private String position;

    @JsonProperty("dept_id")
    @Column(name = "dept_id")
    private Long deptId;
    
    // This field is for joining and displaying, not a direct column in employee table
    @Transient
    @JsonProperty("dept_name")
    private String deptName;

    private Double salary;

    private String status;

    private String phone;

    private String email;

    @JsonProperty("join_date")
    @Column(name = "join_date")
    @Temporal(TemporalType.DATE)
    private Date joinDate;

    @JsonProperty("create_time")
    @Column(name = "create_time", updatable = false)
    @Temporal(TemporalType.TIMESTAMP)
    private Date createTime;

    @JsonProperty("update_time")
    @Column(name = "update_time")
    @Temporal(TemporalType.TIMESTAMP)
    private Date updateTime;
} 

File: src/types/employee.ts
import type { ApiResponse, BaseFields } from './common'

// 后端返回的员工数据类型
export interface EmployeeItemResponse {  // 改名为 EmployeeItemResponse
  id: number
  emp_id: string
  name: string
  gender: string
  age: number
  dept_name: string
  position: string
  salary: number
  status: string
  phone: string | null
  email: string | null
  join_date: string
  create_time: string
}

// 前端使用的员工数据类型
export interface EmployeeItem {
  id: number
  empId: string
  name: string
  gender: string
  age: number
  deptName: string
  position: string
  salary: number
  status: string
  phone?: string
  email?: string
  joinDate: string
  createTime: string
}

// 员工表单数据类型
export interface EmployeeFormData {
  id?: number
  empId: string
  name: string
  gender: string
  age: number
  position: string
  deptName: string
  salary: number
  status: string
  phone?: string
  email?: string
  joinDate: string
}

// 部门数据类型
export interface DeptResponse extends BaseFields {
  dept_name: string
  manager: string
  member_count: number
  description: string | null
}

export interface DeptItem {
  id: number
  deptName: string  
  manager: string
  memberCount: number
  description?: string
  createTime: string
}

// API 响应类型
export type ApiEmployeeResponse = ApiResponse<EmployeeItemResponse[]>
export type ApiDeptResponse = ApiResponse<DeptResponse[]>